modules/private-types/src/types.ts

```diff
@@ -0,0 +1,336 @@
+// TODO: why interfaces and not classes?
+
+export interface BulkPerTransferFeeLimit {
+	currency: Currency;
+	amount: Money;
+}
+
+export interface IAutoAcceptPartyOption {
+	enabled: boolean;
+}
+
+export interface IAutoAcceptQuoteOption {
+	enabled: boolean;
+	perTransferFeeLimits: BulkPerTransferFeeLimit;
+}
+
+export interface IBulkTransferOptions {
+	onlyValidateParty: boolean;
+	autoAcceptParty: IAutoAcceptPartyOption;
+	autoAcceptQuote: IAutoAcceptQuoteOption;
+	skipPartyLookup: boolean;
+	synchronous: boolean;
+}
+
+export interface IExtensionItem {
+	key: string;
+	value: string;
+}
+
+export interface BulkExtensionList {
+	items: Array<IExtensionItem>;
+}
+
+export interface IBulkTransferRequest {
+	options: IBulkTransferOptions;
+	extensionList: BulkExtensionList;
+}
+
+export interface IExtensionList {
+	items: Array<IExtensionItem>;
+}
+
+export interface IFspId {
+	fspId: string;
+}
+
+export interface IPartyIdInfo {
+	partyIdType: IIdType;
+	partyIdentifier: string;
+	partySubIdOrType: string;
+	fspId: IFspId;
+	extensionList: IExtensionList;
+}
+
+export interface IMerchantClassificationCode {
+	merchantClassificationCode: string;
+}
+
+export interface PartyName {
+	partyName: string;
+}
+
+
+export interface IPartyComplexName {
+	firstName: PartyName;
+	middleName: PartyName;
+	lastName: PartyName;
+}
+
+export interface IDateShort {
+	dateShort: string;
+}
+
+export interface IPersonalInfo {
+	complexName: IPartyComplexName;
+	dateOfBirth: IDateShort;
+}
+
+// TODO verify
+export interface IPartyRequest {
+	partyIdInfo: IPartyIdInfo;
+	merchantClassificationCode: IMerchantClassificationCode;
+	name: PartyName;
+	personalInfo: IPersonalInfo;
+}
+
+export interface IIndividualTransfers {
+	id: string;
+	request: IBulkTransferRequest;
+	status: IBulkTransferStatus;
+	batchId: string;
+	partyRequest: IPartyRequest; // TODO verify
+	partyResponse: IPartyRequest; // TODO Define Party Response...once we have identified what it should be
+	acceptParty: boolean;
+	acceptQuote: boolean;
+}
+
+export interface BulkBatch {
+	id: string;
+	status: IBulkTransferStatus;
+	bulkQuoteId: string;
+	bulkTransferId: string;
+}
+
+export interface IBulkTransferEntity {
+	bulkTransactionId: string;
+	bulkHomeTransactionID: string | null;
+	request: IBulkTransferRequest;
+	individualtransfers: IIndividualTransfers;
+	status: IBulkTransferStatus;
+	bulkBatch: BulkBatch;
+	partyLookupTotalCount: number;
+	partyLookupSuccessCount: number;
+	partyLookupFailedCount: number;
+	bulkQuotesTotalCount: number;
+	bulkQuotesSuccessCount: number;
+	bulkQuotesFailedCount: number;
+	bulkTransfersTotalCount: number;
+	bulkTransfersSuccessCount: number;
+	bulkTransfersFailedCount: number;
+}
+
+// TODO: check name and status enums.
+export enum IBulkTransferStatus {
+	ACTIVE = "RECEIVED",
+	DELETED = "DISCOVERY_PROCESSING",
+	AGREEMENT_PROCESSING = "AGREEMENT_PROCESSING",
+	TRANSFER_PROCESSING = "TRANSFER_PROCESSING"
+}
+
+export enum IIdType {
+	MSISDN = "MSISDN",
```

Comment:
Why only these types? 

All types that are allowable should be adhering to the FSPIOP Specification.

Another option is to not make this specific.